docs/developer.md

```diff
@@ -0,0 +1,99 @@
+# DEVELOPER GUIDE --- WIP
+
+## Contributing
+// TODO(user): Add detailed information on how you would like others to contribute to this project
+
+### How it works
+This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/).
+
+It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/)
+which provides a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster.
+
+
+## Local development
+
+### Prerequisites
+
+You need ready to use **kubectl** and **minikube** [tools](https://kubernetes.io/docs/tasks/tools/) as a prerequisite.
+
+Get your copy of Backstage Operator from GitHub:
+```sh
+git clone https://github.com/janus-idp/operator
+```
+
+### Local Tests
+
+Including unit testing (starting from v 0.02) and Kubernetes intergation testing ([envtest](https://book.kubebuilder.io/reference/envtest.html))
+
+```sh
+make test
+```
+
+will run both of them  
+
+### Test on the local cluster
+
+You’ll need a Kubernetes cluster to run against. You can use [Minikube](https://minikube.sigs.k8s.io/docs/) or [KIND](https://sigs.k8s.io/kind) to get a local cluster for testing, or run against a remote cluster.
```

Comment:
Should we link to https://kubernetes.io/docs/tasks/tools/ instead, since that's what you have above for the prereqs?